account_asset_management/models/account_move.py

```diff
@@ -151,13 +151,14 @@ class AccountMoveLine(models.Model):
 
     @api.onchange("account_id")
     def _onchange_account_id(self):
-        self.asset_profile_id = self.account_id.asset_profile_id
+        if self.account_id.asset_profile_id:
+            self.asset_profile_id = self.account_id.asset_profile_id
         super()._onchange_account_id()
 
     @api.onchange("asset_profile_id")
     def _onchange_asset_profile_id(self):
-        self.account_id = self.asset_profile_id.account_asset_id
-        super()._onchange_account_id()
+        if self.asset_profile_id.account_asset_id:
```

Comment:
AFAIK, this field is required, isn't it?

Reply:
Actually if we installed account_global_discount + account_asset_management test fail in https://github.com/OCA/account-invoicing/blob/13.0/account_global_discount/tests/test_global_discount.py#L111-L118 (`account_id` is required)
IMO reason it's `_onchange_account_id` that call to `_onchange_asset_profile_id` and define `account_id` in invoice line as empty.